Fatras/include/ActsFatras/Physics/PhotonConversion/PhotonConversion.hpp

```diff
@@ -0,0 +1,311 @@
+    }
+  } while (greject < rndmEngine(generator));
+  //  End of eps sampling
+  return eps * childEnergyScaleFactor;
+}
+
+template <typename generator_t>
+Particle::Vector3 PhotonConversion::childDirection(
+    generator_t& generator, const Particle& particle) const {
+  /// This method is based upon the Athena class PhotonConversionTool
+
+  // Following the Geant4 approximation from L. Urban
+  // the azimutal angle
+  Scalar theta = electronMass / particle.energy();
+
+  std::uniform_real_distribution<Scalar> uniformDistribution{0., 1.};
+  const Scalar u = -std::log(uniformDistribution(generator) *
+                             uniformDistribution(generator)) *
+                   1.6;
+
+  theta *= (uniformDistribution(generator) < 0.25)
+               ? u
+               : u * 1. / 3.;  // 9./(9.+27) = 0.25
+
+  // draw the random orientation angle
+  const auto psi =
+      std::uniform_real_distribution<double>(-M_PI, M_PI)(generator);
+
+  Acts::Vector3 direction = particle.unitDirection();
+  // construct the combined rotation to the scattered direction
+  Acts::RotationMatrix3 rotation(
+      // rotation of the scattering deflector axis relative to the reference
+      Acts::AngleAxis3(psi, direction) *
+      // rotation by the scattering angle around the deflector axis
+      Acts::AngleAxis3(theta, Acts::makeCurvilinearUnitU(direction)));
+  direction.applyOnTheLeft(rotation);
+  return direction;
+}
+
+std::array<Particle, 2> PhotonConversion::generateChildren(
+    const Particle& photon, Scalar childEnergy,
+    const Particle::Vector3& childDirection) const {
+  // Calculate the child momentum
+  const Scalar massChild = electronMass;
+  const Scalar momentum1 =
+      sqrt(childEnergy * childEnergy - massChild * massChild);
+
+  // Use energy-momentum conservation for the other child
+  const Particle::Vector3 vtmp =
+      photon.fourMomentum().template segment<3>(Acts::eMom0) -
+      momentum1 * childDirection;
+  const Scalar momentum2 = vtmp.norm();
+
+  // Build the particles and store them
+  std::array<Particle, 2> children;
+
+  Particle child1 = Particle(photon.particleId().makeDescendant(0),
+                             static_cast<Acts::PdgParticle>(Acts::eElectron))
+                        .setPosition4(photon.fourPosition())
+                        .setDirection(childDirection)
+                        .setAbsoluteMomentum(momentum1)
+                        .setProcess(ProcessType::ePhotonConversion);
+  children[0] = std::move(child1);
+
+  Particle child2 = Particle(photon.particleId().makeDescendant(1),
+                             static_cast<Acts::PdgParticle>(-Acts::eElectron))
+                        .setPosition4(photon.fourPosition())
+                        .setDirection(childDirection)
+                        .setAbsoluteMomentum(momentum2)
+                        .setProcess(ProcessType::ePhotonConversion);
+  children[1] = std::move(child2);
+
+  return children;
```

Comment:
```suggestion
  std::array<Particle, 2> children = {
    Particle(photon.particleId().makeDescendant(0), Acts::eElectron)
      .setPosition4(photon.fourPosition())
      .setDirection(childDirection)
      .setAbsoluteMomentum(momentum1)
      .setProcess(ProcessType::ePhotonConversion),
    Particle(photon.particleId().makeDescendant(1), Acts::ePositron)
      .setPosition4(photon.fourPosition())
      .setDirection(childDirection)
      .setAbsoluteMomentum(momentum2)
      .setProcess(ProcessType::ePhotonConversion),
  };
  return children;
```
You can initialize the elements of the array directly without having to move them.

Reply:
I was afraid about the readability but changed it